Lib/test/test_c_locale_coercion.py

```diff
@@ -301,6 +319,11 @@ def _check_c_locale_coercion(self,
                 # See https://bugs.python.org/issue30672 for discussion
                 if locale_to_set == "POSIX":
                     continue
+                elif locale_to_set == "" and DEFAULT_LC_CTYPE == "C.UTF-8":
```

Comment:
The condition is too restrictive:
 * When `locale_to_set` is `""` and  a locale envt variable exists in `extra_vars` with a non empty string (for example in `test_LC_ALL_set_to_C`), the subtest may be run.
* It seems that when  `C_LOCALE_STREAM_ENCODING` is set to `"utf-8"` and `coerce_c_localecoerce_c_locale` is not `"warn"` then the test should pass.